### PR TITLE
Make jit.trace check_inputs behavior consistent with doc

### PR DESCRIPTION
**Pull request by ssnl**

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Review by jamesr66a — CHANGES_REQUESTED** [review]
This seems to make `check_trace` no-op? I don't think that's what we want. That flag is used extensively (grep `test/test_jit.py` for it) and is required if tracing is to be used with any network with nodeterministic ops, e.g. dropout

**Comment by jamesr66a** [comment]
Additionally, `check_trace` is used to disable checking in the inner call to `trace()` during checking itself. In CI you have infinite recursion happening: https://circleci.com/gh/pytorch/pytorch/894883?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link&fullLogs=true

This is TP failure and we should not merge this.

**Comment by SsnL** [comment]
@jamesr66a But this doesn't make `check_inputs` a no-op? Could you elaborate what you mean?

**Comment by SsnL** [comment]
@jamesr66a And just in the contrary, this makes `check_inputs` does what it is supposed to do, rather than no-op in certain cases. And you can still disable the check by passing in `[]`. I'm really just following the doc behavior.

**Comment by jamesr66a** [comment]
@SsnL `check_trace` and `check_inputs` are two *separate* flags, both documented in the comment. The first is a boolean flag that can be used to disable checking, the second is a List of inputs. By deleting `if check_trace:` you're made that flag no-op

File: torch/jit/__init__.py [hunk 1]
```diff
@@ -663,11 +664,10 @@ def trace(func,
                                      var_lookup_fn, _force_outplace)
 
     # Check the trace against new traces created from user-specified inputs
-    if check_trace:
-        if check_inputs is not None:
-            _check_trace(check_inputs, func, executor_options, module, check_tolerance, _force_outplace)
-        else:
-            _check_trace([example_inputs], func, executor_options, module, check_tolerance, _force_outplace)
```
Comment (SsnL) [inline torch/jit/__init__.py]: @jamesr66a And the original code made zero sense. This line will never be executed.

Comment (jamesr66a) [reply torch/jit/__init__.py]: @SsnL these are two different names, `check_inputs` and `check_trace`

**Comment by SsnL** [comment]
Oh I see. Yeah you are correct. I'll close this.

**Comment by jamesr66a** [comment]
I'd accept a PR to make the two flags more distinctly named, though. I don't blame you for making this mistake, they look the same at first glance! :)

**Comment by SsnL** [comment]
Sorry for the mistake. They also kind of have overlapping effects. But this is already public API, so I don't know how much we can change their names/effects. I'll leave them this way.

**Pull request closed without merging**